lib/impure/rdstdin.nim

```diff
@@ -100,12 +100,17 @@ when defined(Windows):
     stdout.write "\n"
 
 else:
-  import linenoise, termios
+  import linenoise, termios, posix
 
   proc readLineFromStdin*(prompt: string): TaintedString {.
                           tags: [ReadIOEffect, WriteIOEffect].} =
+    errno = 0
     var buffer = linenoise.readLine(prompt)
     if isNil(buffer):
+      # linenoise eats SIGINT and sets errno to EAGAIN, therefore reraise SIGINT.
```

Comment:
I did not come up with ``errno`` not did I come up with this API. I also did not come up with the idea to use linenoise. All I did was to make this API work with what linenoise provides. Don't blame me for the bad design of the linenoise library.